tests/components/wemo/test_fan.py

```diff
@@ -130,3 +135,47 @@ async def test_fan_set_humidity_service(
         blocking=True,
     )
     pywemo_device.set_humidity.assert_called_with(expected)
+
+
+@pytest.mark.parametrize(
+    "percentage,expected_fan_mode",
+    [
+        (0, FanMode.Off),
+        (10, FanMode.Minimum),
+        (30, FanMode.Low),
+        (50, FanMode.Medium),
+        (70, FanMode.High),
+        (100, FanMode.Maximum),
+    ],
+)
+async def test_fan_set_percentage(
+    hass, pywemo_device, wemo_entity, percentage, expected_fan_mode
+):
+    """Verify set_percentage works properly through the entire range of FanModes."""
+    assert await hass.services.async_call(
+        FAN_DOMAIN,
+        SERVICE_SET_PERCENTAGE,
+        {ATTR_ENTITY_ID: [wemo_entity.entity_id], ATTR_PERCENTAGE: percentage},
+        blocking=True,
+    )
+    pywemo_device.set_state.assert_called_with(expected_fan_mode)
+
+
+class TestInitialFanMode:
+    """Test that the FanMode is set to High when turned on the first time."""
+
+    @pytest.fixture
+    def pywemo_device(self, pywemo_device):
+        """Set the FanMode to off initially."""
+        pywemo_device.fan_mode = FanMode.Off
+        yield pywemo_device
+
```

Comment:
Usually we don't put tests inside of a class

Reply:
I've done this to scope the `pywemo_device` fixture so it only impacts this one test. The fan_mode needs to be Off only for this one test case and not the others. Is there an alternative way to do this that would fit better?

Reply:
If you'd prefer, I can delete this test also. It isn't adding much additional coverage.

Reply:
One alternative would be to move it into another file, but that seems like it would go against the goal of keeping these less complex. 

I can't think of a better solution without doing some major refactoring to how the fixtures are setup.